### ISSUE TITLE
Proxy headers

### ISSUE THREAD

**Issue by theArina** [apify/proxy-chain#121]
How can I add proxy headers to upstream proxy ?

**Comment by jancurn** [comment]
I assume you'd like to add extra headers to the `HTTP CONNECT` method, right? Currently, this is not possible. See `HandlerTunnelChain` implementation, PRs welcome.

**Comment by theArina** [comment]
@jancurn, I'll take a look at this, thanks.

**Comment by dmitryshelomanov** [comment]
@theArina has solution ?

**Comment by theArina** [comment]
@dmitryshelomanov, yeah, we made a [fork.](https://github.com/petr-vysotskiy/proxy-chain) take a look, but we also added some enrichments that we needed.

**Comment by dmitryshelomanov** [comment]
@theArina any docs ?

**Comment by theArina** [comment]
@dmitryshelomanov, for a fork ?

**Comment by theArina** [comment]
@dmitryshelomanov, well, shortly you just kinda return this
`{ requestAuthentication: false, upstreamProxyUrl, proxyHeaders };`
instead of this
`{ requestAuthentication: false, upstreamProxyUrl };`
from the **prepareRequestFunction** 
(but not necessarily)

**Comment by bliuchak** [comment]
Closing this as resolved.
